src/go/k8s/tests/e2e/external-connectivity/01-assert.yaml

```diff
@@ -0,0 +1,67 @@
+apiVersion: apps/v1
```

Comment:
is this really testing something e2e? 🤔  other than the fact that the resources are applied

Reply:
I can argue that not all tests created in kuttl are end-to-end 😄 . To be more concrete. 

What you would like to see?

As a context:
On Mac there is not easy way to expose kind cluster to the host system. There are possible workarounds like:

> Extra port mappings can be used to port forward to the kind nodes. This is a cross-platform option to get traffic into your kind cluster.
>
> With docker on Linux you can simply send traffic to the node IPs from the host without this, but to cover macOS and Windows you'll want to use these. 

REF https://kind.sigs.k8s.io/docs/user/configuration#extra-port-mappings

The downside is that ports need to be known before kind cluster is created.

I will try to create the same produce-consune e2e tests using this _extra port mappings_

Reply:
so we could curl from kuttl, right? But you're saying that test would not work on mac 🤔 hmm I see, yeah, I am not happy we don't have this tested really but I won't block on this

Reply:
I have workaround for that need some more time to implement the test. 

The gist for it is that using extra port mapping I can expose one port to the mac host. I will create Cluster CR and then assert all the goodies. Then overwrite the NodePort to see if it can be adjusted/changed in runtime. Then run `go test` which will use sarama to create topic, produce message and consume message (as I didn't in one of my PRs).

I'm not sure if Redpanda can change in the runtime advertised kafka api, because the NodePort will change.